enhancements/olm/simplify-apis.md

```diff
@@ -0,0 +1,523 @@
+
+A Cluster Admin can generate `ServiceAccounts` with different installation privileges as needed; e.g. namespace list, pod create, etc. To allow a user to reference a `ServiceAccount`, a Cluster Admin can bind a `Role/ClusterRole` with the `use` custom verb to the user.
+
+This behavior will be enforced by a `ValidatingAdmissionWebhook`. In order to configure serving certs for the webhook
+
+- in OLM's upstream deployment, bootstrap and use [cert-manager](https://docs.cert-manager.io/en/latest/tasks/issuers/index.html) with self-signed certs by default
+- in OLM's downstream deployment (OKD), use the [service-ca-operator](https://github.com/openshift/service-ca-operator)
+- (optional) at some point support validation/mutating webhooks in OLM and self-host the validating admission server
+
+#### Component Selection
+
+Components can be associated with an `Operator` by a label following the key convention `operators.coreos.com/v2/operator/<operator-name>`. This choice of convention has the following benefits:
+
+- using a unique label key helps avoid collisions
+- using a deterministic label key lets users know the exact label used in advance
+
+The resolved label selector for an `Operator` will be surfaced in the `status.components.matchLabels` field of its status.  
+
+```yaml
+status:
+  components:
+    matchLabels:
+      operators.coreos.com/v2/operators/my-operator: ""
+```
+
+**Note:** *Both namespace and cluster scoped resources are gathered using this label selector. Namespace scoped components are selected across all namespaces.*
+
+Once associated with an `Operator`, a component's reference will be listed in the `status.components.resource` field of that `Operator`. Component references will also be enriched with abnormal status conditions relevant to the operator. These conditions should follow [k8s status condition conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) and in some cases may be copied directly from the component status.
+
+```yaml
+status:
+  components:
+    matchLabels:
+        operators.coreos.com/v2/operators/my-operator: ""
+    resources:
+    - kind: ClusterServiceVersion
+      namespace: my-ns
+      name: plumbus.v2.0.0-alpha
+      uid: d70a53b5-d06a-11e9-821f-9a2e3b9d8156
+      apiVersion: operators.coreos.com/v1alpha1
+      resourceVersion: 109811
+      conditions:
+      - type: Installing
+        status: True
+        reason: AllPreconditionsMet
+        message: deployment rolling out
+        lastTransitionTime: "2019-09-16T22:26:29Z"
+```
+
+Not all components will need to have status conditions surfaced. Initially, it should be sufficient to add conditions for:
```

Comment:
Not every resource has status conditions and we'll most likely need to do a mapping when we implement this (since under the cover they are different golang types). Other than that, I was thinking that it didn't make sense to have conditions for resources like `ServiceAccounts` and RBAC, etc... That being said, we can always add more as we find we need them.